test/distributed/test_c10d.py

```diff
@@ -4639,6 +4640,48 @@ def test_broadcast_coalesced_gloo_cpu(self):
         for root_rank in ranks:
             self._test_broadcast_coalesced(process_group, device, root_rank)
 
+    @requires_gloo()
+    def test_pass_gloo_options(self):
+        pg_opts = c10d.ProcessGroupGloo.Options()
+        pg_opts.devices = [create_device(interface=LOOPBACK)]
```

Comment:
Can we somehow test that we appropriately set devices if the user does not set it themselves?

Reply:
Yeah that's also something I would like to test, but I don't actually know how to test the devices/threads attribute creation yet, given that we don't hold a `options` field in ProcessGroupGloo and there's currently no way for us to access the protected fields in python (from the [python bindings](https://github.com/pytorch/pytorch/blob/master/torch/csrc/distributed/c10d/init.cpp#L1125) currently, we don't have access to fields like `context_` and `threads_`). So unless we want to expose them in python or add a new field in ProcessGroupGloo and bind it in python, we couldn't know if we set up the devices or threads?